pkg/configauditreport/plugin.go

```diff
@@ -15,9 +13,18 @@ import (
 // first step to separate generic workloads discovery code and Polaris
 // implementation details.
 type Plugin interface {
-	GetScanJobSpec(workload kube.Object, obj client.Object, gvk schema.GroupVersionKind) (corev1.PodSpec, []*corev1.Secret, error)
 
-	GetContainerName() string
+	// GetScanJobSpec describes the pod that will be created by Starboard when
+	// it schedules a Kubernetes job to scan the specified workload client.Object.
+	// The plugin might return zero to many v1.Secret objects which will be
+	// created by Starboard and associated with the scan job.
+	GetScanJobSpec(obj client.Object) (corev1.PodSpec, []*corev1.Secret, error)
```

Comment:
In the beginning `kube.Object` and `schema.GroupVersionKind` args where used to integrate with Polaris. After exploring other tools such as Conftest I believe it's more convenient to pass the whole `client.Object`. We can always infer `kube.Object` and GVK from `client.Object`.